lib/livebook_web/live/hub/edit/fly_component.ex

```diff
@@ -103,15 +110,192 @@ defmodule LivebookWeb.Hub.Edit.FlyComponent do
+            class="icon-button text-red-500 hover:text-red-800"
+          >
+            <.remix_icon icon="delete-bin-line" class="align-middle mr-1" />
+            <span>Delete</span>
+          </button>
+        </div>
+      </div>
     </div>
     """
   end
 
+  defp secret_modal(assigns) do
+    ~H"""
+    <.modal id="secret-modal" class="w-full max-w-sm">
+      <div class="p-6 pb-4 max-w-4xl flex flex-col space-y-5">
+        <h3 class="text-2xl font-semibold text-gray-800">
+          Add secret
+        </h3>
+        <div class="flex-col space-y-5">
+          <p class="text-gray-700">
+            Enter the secret name and its value.
+          </p>
+          <.form
+            id="secret-form"
+            let={f}
+            for={:secret}
+            phx-submit={@on_save |> JS.push("save")}
+            phx-change="validate"
+            autocomplete="off"
+            phx-target={@myself}
+          >
+            <div class="flex flex-col space-y-4">
+              <div>
+                <div class="input-label">
+                  Key <span class="text-xs text-gray-500">(alphanumeric and underscore)</span>
+                </div>
+                <%= text_input(f, :key,
+                  value: @data["key"],
+                  class: "input",
+                  placeholder: "secret key",
+                  autofocus: true,
+                  aria_labelledby: "secret-key",
+                  spellcheck: "false"
+                ) %>
+              </div>
+              <div>
+                <div class="input-label">Value</div>
+                <%= text_input(f, :value,
+                  value: @data["value"],
+                  class: "input",
+                  placeholder: "secret value",
+                  aria_labelledby: "secret-value",
+                  spellcheck: "false"
+                ) %>
+              </div>
+            </div>
+            <%= submit("Add Secret",
+              class: "mt-5 button-base button-blue",
+              phx_disable_with: "Adding...",
+              disabled: not @valid?
+            ) %>
+          </.form>
+        </div>
+      </div>
+    </.modal>
+    """
+  end
+
   @impl true
   def handle_event("randomize_color", _, socket) do
     handle_event("validate", %{"fly" => %{"hub_color" => HexColor.random()}}, socket)
   end
 
+  def handle_event("edit", %{"secret" => %{"name" => name}}, socket) do
+    {:noreply, assign(socket, operation: :edit, secret_data: %{"key" => name})}
+  end
+
+  def handle_event("delete", %{"secret" => %{"name" => key}}, socket) do
+    case FlyClient.delete_secrets(socket.assigns.hub, [key]) do
+      {:ok, _} ->
+        {:noreply,
+         socket
+         |> put_flash(:success, "Secret deleted successfully")
+         |> push_redirect(to: Routes.hub_path(socket, :edit, socket.assigns.hub.id))}
+
+      {:error, _} ->
+        {:noreply,
+         socket
+         |> put_flash(:error, "Failed to delete secret")
```

Comment:
```suggestion
         |> put_flash(:error, "Failed to delete environment variable")
```

Please change the other messages accordingly. :) All of those are environment variables.